frontends/election-manager/src/screens/write_ins_screen.tsx

```diff
@@ -1,15 +1,88 @@
-import React from 'react';
+import React, { useContext, useState } from 'react';
+
+import { Button, Modal } from '@votingworks/ui';
+import {
+  CandidateContest,
+  getPartyAbbrevationByPartyId,
+} from '@votingworks/types';
 
 import { NavigationScreen } from '../components/navigation_screen';
 import { Prose } from '../components/prose';
+import { WriteInsTranscriptionScreen } from '../components/write_ins_transcription_screen';
+import { AppContext } from '../contexts/app_context';
 
 export function WriteInsScreen(): JSX.Element {
+  const { castVoteRecordFiles, electionDefinition } = useContext(AppContext);
+  const election = electionDefinition?.election;
+  const castVoteRecordFileList = castVoteRecordFiles.fileList;
+  const hasCastVoteRecordFiles =
+    castVoteRecordFileList.length > 0 || !!castVoteRecordFiles.lastError;
+  const contestsWithWriteIns = election?.contests.filter(
+    (contest): contest is CandidateContest =>
+      contest.type === 'candidate' && contest.allowWriteIns
+  );
+
+  // Aggregate write-in count by contest
+  const writeInCountsByContest = new Map();
```

Comment:
This should have type parameters, otherwise the `key` and `value` are both `any`:

```suggestion
  const writeInCountsByContest = new Map<ContestId, number>();
```